harness/testIntl.js

```diff
@@ -771,6 +771,65 @@ function isCanonicalizedStructurallyValidLanguageTag(locale) {
 }
 
 
+/**
+ * Returns an array of error cases handled by CanonicalizeLocaleList().
+ */
+function getInvalidLocaleArguments() {
+  function CustomError() {}
+
+  var topLevelErrors = [
+    // fails ToObject
+    [null, TypeError],
+
+    // fails Get
+    [{ get length() { throw new CustomError(); } }, CustomError],
+
+    // fail ToLength
+    [{ length: Symbol.toPrimitive }, TypeError],
+    [{ length: { get [Symbol.toPrimitive]() { throw new CustomError(); } } }, CustomError],
+    [{ length: { [Symbol.toPrimitive]() { throw new CustomError(); } } }, CustomError],
+    [{ length: { get valueOf() { throw new CustomError(); } } }, CustomError],
+    [{ length: { valueOf() { throw new CustomError(); } } }, CustomError],
+    [{ length: { get toString() { throw new CustomError(); } } }, CustomError],
+    [{ length: { toString() { throw new CustomError(); } } }, CustomError],
```

Comment:
For custom errors you may also use Test262Error, no need to create a new function. That's optional.

Reply:
I previously allowed a PR that used this `CustomError`, with the expectation that I'd fix it to use `Test262Error` and I didn't get to it, sorry. We can allow this here and then I'll fix them all today.

Reply:
> That's optional.

as I said. It doesn't hurt the tests in anyway. Consistency would be the only achievement from this change.

Reply:
The docs say

> Test262Error(message) | constructor for an error object that indicates a test failure

which makes it seem inappropriate for my purpose. If you think it should be used for cases like this, can you update the docs as well?